python/cog/server/redis_queue.py

```diff
@@ -274,6 +286,13 @@ def handle_message(
 
             while self.runner.is_processing():
                 # TODO: restructure this to avoid the tight CPU-eating loop
+                if should_cancel():
```

Comment:
Does this add any latency by checking for this on each iteration? Not important, but might be something for future optimization.